lib/collection/src/wal_delta.rs

```diff
@@ -81,18 +92,19 @@ impl RecoverableWal {
     pub async fn update_cutoff(&self, recovery_point: &RecoveryPoint) {
         let mut highest_clocks = self.highest_clocks.lock().await;
         let mut cutoff_clocks = self.cutoff_clocks.lock().await;
-        recovery_point
-            .clock_tag_iter()
-            .filter(|clock_tag| clock_tag.clock_tick >= 1)
-            .for_each(|mut clock_tag| {
-                clock_tag.clock_tick -= 1; // ToDO: get rid of
-                                           // Advance highest clocks we've seen
-                highest_clocks.advance_clock(clock_tag);
-
-                // Advance the cutoff point
-                // A recovery point has every clock one higher, here we decrease it by one again
-                cutoff_clocks.advance_clock(clock_tag);
-            });
+
+        recovery_point.clock_tag_iter().for_each(|mut clock_tag| {
+            // A recovery point has every clock one higher, here we decrease it by one again
+            //
+            // TODO: Refactor `RecoveryPoint` to transfer *current* clocks instead of *next* clocks!
+            clock_tag.clock_tick -= 1;
+
+            // Advance highest clocks we've seen
+            highest_clocks.advance_clock(clock_tag);
+
+            // Advance the cutoff point
+            cutoff_clocks.advance_clock(clock_tag);
```

Comment:
Nitpick, but after reading this a bunch of times I don't think we have to be overly explicit.

```suggestion
            clock_tag.clock_tick -= 1;
            highest_clocks.advance_clock(clock_tag);
            cutoff_clocks.advance_clock(clock_tag);
```